lstmcpipe/workflow_management.py

```diff
@@ -51,146 +52,45 @@ def batch_r0_to_dl1(input_dir, conf_file, prod_id, particles_loop, source_env, g
-                input_dir.format(particle),  # Input dir needs particle gamma w/o off
-                config_file=conf_file,
-                particle=_particle,  # However _particle will contain the gamma_off
-                prod_id=prod_id,
-                flag_full_workflow=True,
-                source_environment=source_env
-            )
-
-            # Create dictionary : jobid to full log information, and
-            #  the inverse dictionary, particle to the list of all the jobids of that same particle
-            full_log['log_all_job_ids'].update(log)
-            full_log[_particle] = ','.join(jobids_by_particle)
-            all_jobids_from_r0_dl1_stage.append(full_log[_particle])  # Create a list with particles elements
-
-            for jid in jobids_by_particle:
-                debug_log[jid] = f'{_particle} job from r0_to_dl1'
-
-    all_jobids_from_r0_dl1_stage = ','.join(all_jobids_from_r0_dl1_stage)  # Create a string to be directly passed
-
-    print("\n ==== END {} ==== \n".format('batch r0_to_dl1_workflow'))
-
-    return full_log, debug_log, all_jobids_from_r0_dl1_stage  # ids_by_particle_ok
-
-
-def batch_r0_to_dl1_rta(input_dir, conf_file_rta, prod_id, particles_loop, conf_file_lst, gamma_offsets=None):
-    """
-    Function to batch the r0_to_dl1 jobs by particle type, using the HiPeRTA code. Files in input_dir MUST had been
-     previously converted to *.h5
-
-    It will also create, arrange and return a dictionary with all the log of this stage.
-
-    Parameters
-    ----------
-    input_dir : str
-        Path to the R0 (h5 !) files
-    conf_file_rta : str
-        Path to a HiPeRTA configuration file. If none is given, a standard configuration is applied
-    prod_id : str
-        Production ID. If None, _v00 will be used, indicating an official base production. Default = None.
-    particles_loop : list
-        list with the particles to be processed. Takes the global variable ALL_PARTICLES
-    conf_file_lst : str
-        Path to a lstchain configuration. JUST to be copied at the same time as the rta_config to `/running_analysis/`.
-    gamma_offsets : list
-
-    Returns
-    -------
-    full_log : dict
-        Dictionary of dictionaries containing the full log of the batched jobs (jobids as keys) as well as the
-        4 more keys (one by particle) with all the jobs associated with each particle.
-
-    debug_log : dict
-            dictionary containing minimum information - jobids -  for log_reduced.txt
-
-    all_jobids_from_r0_dl1_stage : str
-        string, separated by commas, containing all the jobids of this stage
-    """
-    full_log = {'log_all_job_ids': {}}
-    debug_log = {}
-    all_jobids_from_r0_dl1_stage = []
+        for jid in jobids_by_particle:
+            debug_log[jid] = f'{_particle} job from r0_to_dl1'
 
-    print("\n ==== START {} ==== \n".format('HiPeRTA_r0_to_dl1_workflow'))
+        return log, jobids_by_particle
```

Comment:
However, `start_r0` should not return nothing then. Each time time `start_r0` is invoked, the `full_log` and the `debug_log` (defined before) will be updated.

Reply:
Is it not supposed to update the whole log? Thats what it did before. It defined the logs once before the loop and updated it in every step.

Anyway, I'll change the function as Max noted above. I don't think its harder to reason about personally, but given that we are discussing it quite a bit, I am probably in the wrong here :D

Reply:
`log` and `jobids_by_particle` come from `r0_to_dl1`. `log` should update `full_log` and `jobids_by_particle` `debug_log` respectively. So `start_r0` should return nothing.

Reply:
Ah yes, the return values are not needed the way its coded right now.
Its either return nothing, update in function or return values, update outside of the function.
With the second case being the easier one to understand according to Max.